src/akamai/js/akamaiHD_flash.js

```diff
@@ -449,8 +490,97 @@ require("../../../html5-common/js/utils/constants.js");
     var raiseSizeChanged = function(event) {
     };
 
-    // Receives a callback from Flash
+    // Receives a callback from Flash 
     onCallback = _.bind(function(data) {
+      OO.log("[AKAMIHD]:onCallback: ", data);
+      var eventtitle =" ";
+
+      for(var key in data) {
+        if (key == "eventtype") {
+             eventtitle = data[key];
+        }
+        else if (key =="eventObject") {
+              eventData = data[key];
+        }
+      }
+      if (eventData != null) {
+        for (var item in eventData)
+        {
+          if (item == "currentTime") {
+                currentTime = eventData[item];
+          }
+          else if (item == "buffer") {
+                buffer = eventData[item];
+          }
+          else if (item == "duration") {
+                totalTime = eventData[item];
+          }
+          else if (item == "seekRange_start") {
+                seekRange_start = eventData[item];
+          }
+          else if (item == "seekRange_end") {
+                seekRange_end = eventData[item];
+          }
+        }
+      }
+
+      switch (eventtitle)
+      {
+       case "JSREADY":
+        for (i = 0; i < actionscriptCommandQueue.length; i++) {
```

Comment:
@angieisrani , just for my understanding, are you referring to using something like actionscriptCommandQueue.shift() instead here?